sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -106,6 +107,22 @@ object TypeCoercion {
     case (t1, t2) => findTypeForComplex(t1, t2, findTightestCommonType)
   }
 
+  /**
+   * Finds a wider decimal type between the two supplied decimal types without
+   * any loss of precision.
+   */
+  def findWiderDecimalType(d1: DecimalType, d2: DecimalType): Option[DecimalType] = {
```

Comment:
I'd like to rename it to `findTightestDecimalType`, and add document to say what's the difference between this and `findWiderTypeForDecimal`

Reply:
@cloud-fan Sure.. will do.